tests/test_pulp_clone_repos.py

```diff
@@ -0,0 +1,793 @@
+
+        pulpAdminConfig = PulpAdminConfigClassMock.return_value
+        pulpAdminConfig.name = 'pulp-test'
+        pulpAdminConfig.config_path = 'some_path.json'
+
+        client = pulpAdminConfig.__getitem__.return_value
+        client.__getitem__.return_value = 'admin'
+
+        clone = PulpCloneRepos(self.env, self.release_from, self.release_to, self.data_from_all['repo_family'],
+                               self.data_from_all['variant_uid'], self.data_from_all['arch'],
+                               self.data_from_all['content_category'], skip_repo_check)
+        actual = clone.details(commit=commit)
+
+        # check that class constructor is called once with the value
+        # of env['pdc_server']
+        PDCClientClassMock.assert_called_once_with('pdc-test', develop=True)
+
+        instance.__getitem__.assert_called_with('content-delivery-repos')
+        # check that the API is called twice
+        self.assertEquals(instance.__getitem__.call_count, 2)
+
+        # assert_called_with behaves strangely, and making to
+        # of these won't help. Instead we have to construct
+        # a list of expected call objects, and check that
+        # this list matches the actual call list that happened
+        expected_calls = [
+            call(page_size=0, **expected_query_from),
+            call(page_size=0, **expected_query_to), ]
+
+        self.assertEquals(api._.call_args_list, expected_calls)
+
+        # check that there are exactly 2 queries made
+        self.assertEquals(instance.__getitem__()._.call_count, 2)
+
+        print(actual)
+        print(expected_details)
+
+        # check that the actual details are the same as the expected ones
+        self.assertEquals(expected_details, actual, testMethod.__doc__)
+
+    def check_details_error(self, PDCClientClassMock, PulpAdminConfigClassMock, expected_details,
+                            expected_query_from, expected_query_to, testMethod, query_result_from,
+                            query_result_to, commit, skip_repo_check):
+        """Check the expected and actual."""
+        # get mock instance and configure return value for get_paged
+        instance = PDCClientClassMock.return_value
+        api = instance.__getitem__.return_value
+
+        api._.side_effect = iter([query_result_from, query_result_to])
+
+        pulpAdminConfig = PulpAdminConfigClassMock.return_value
+        pulpAdminConfig.name = 'pulp-test'
+        pulpAdminConfig.config_path = 'some_path.json'
+
+        client = pulpAdminConfig.__getitem__.return_value
+        client.__getitem__.return_value = 'admin'
+
+        clone = PulpCloneRepos(self.env, self.release_from, self.release_to, self.data_from_all['repo_family'],
+                               self.data_from_all['variant_uid'], self.data_from_all['arch'],
+                               self.data_from_all['content_category'], skip_repo_check)
+        '''with self.assertRaises(UsageError) as context:
```

Comment:
Any commented lines are running but I have problem with tox. Can we have meeting after my english lesson (after 14:45)?